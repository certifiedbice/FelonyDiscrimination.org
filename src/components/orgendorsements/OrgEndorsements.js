import React,{Component} from 'react'
import OrgApiService from '../../services/org-api-service';
import './OrgEndorsements.css';

export default class OrgEndorsements extends Component{
    state={setError:null};
    componentDidMount(){
        //set endorsements in state for render
    }
    handleSubmit=e=>{
        e.preventDefault()
        const {endorsement}=e.target;
        // Have to add in userId when user login system is implemented.
        OrgApiService.postOrgEndorsements(/*this.props.userId,*/this.props.orgId,endorsement)
            .catch(this.state.setError);
    }
    render(){
        return(
            <div id='organization-endorsement-form-container'>
                <form id="endorsement-form" name="endorsement-form" aria-label="Endorsement form" onSubmit={this.handleSubmit}>
                    <legend>Submit an Endorsement</legend>
                    <div className="endorsement-form-element-container">
                        <label>Endorse:</label>
                        <select className="endorsement-form-element" name="endorsement" id="endorsement-form-select" required>
                            <option value='positive'>Positive</option>
                            <option value='negative'>Negative</option>
                        </select>
                    </div>
                </form>
            </div>
        );
	}
}