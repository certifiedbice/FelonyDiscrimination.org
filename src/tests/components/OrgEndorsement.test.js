import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import OrgEndorsements from '../../components/orgendorsements/OrgEndorsements';

let orgId='5';

it('renders without crashing',()=>{
    const div=document.createElement('div');
    ReactDOM.render(<OrgEndorsements orgId={orgId}/>,div);
});

it('renders correctly',()=>{
    const tree=renderer
        .create(<OrgEndorsements orgId={orgId}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
});