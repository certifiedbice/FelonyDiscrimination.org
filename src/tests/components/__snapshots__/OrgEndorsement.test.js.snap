// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="organization-endorsement-form-container"
>
  <form
    aria-label="Endorsement form"
    id="endorsement-form"
    name="endorsement-form"
    onSubmit={[Function]}
  >
    <legend>
      Submit an Endorsement
    </legend>
    <div
      className="endorsement-form-element-container"
    >
      <label>
        Endorse:
      </label>
      <select
        className="endorsement-form-element"
        id="endorsement-form-select"
        name="endorsement"
        required={true}
      >
        <option
          value="positive"
        >
          Positive
        </option>
        <option
          value="negative"
        >
          Negative
        </option>
      </select>
    </div>
  </form>
</div>
`;
